util/app/path/path.go

```diff
@@ -27,3 +28,56 @@ func Path(root, path string) (string, error) {
 	}
 	return appPath, nil
 }
+
+const outOfBoundsError string = "Out of bounds symlink found"
+
+// ContainsOutOfBoundsSymlinks determines if basePath contains any symlinks that
+// are absolute or point to a path outside of the basePath.
+func ContainsOutOfBoundsSymlinks(basePath string) (bool, string, error) {
+	found := ""
+	var err error
+	absBasePath, err := filepath.Abs(basePath)
+	if err != nil {
+		return false, "", err
+	}
+	err = filepath.Walk(absBasePath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			// We don't use filepath.EvalSymlinks because it fails without returning a path
+			// if the target doesn't exist.
+			target, err := os.Readlink(path)
+			if err != nil {
+				return err
+			}
+			repoPath, err := filepath.Rel(absBasePath, path)
+			if err != nil {
+				return err
+			}
+			if filepath.IsAbs(target) {
```

Comment:
This is actually kinda funny. While it's true that technically an absolute path could point in-bounds, you'd have to know the path (which should be secret). I like it.  :-)

Reply:
Could this check move up above the `filepath.Rel` check, since it doesn't depend on that var?

Reply:
It does, it's setting `found = repoPath` inside the `IsAbs` check, `repoPath` comes from `filepath.Rel`